SmartDeviceLink/SDLIAPSession.m

```diff
@@ -51,9 +66,16 @@ - (BOOL)start {
 
         strongSelf.streamDelegate.streamErrorHandler = [self streamErroredHandler];
         strongSelf.streamDelegate.streamOpenHandler = [self streamOpenedHandler];
-
-        [strongSelf startStream:weakSelf.easession.outputStream];
-        [strongSelf startStream:weakSelf.easession.inputStream];
+        if (!self.isDataSession) {
+            [strongSelf startStream:weakSelf.easession.outputStream];
+            [strongSelf startStream:weakSelf.easession.inputStream];
+        } else {
+            strongSelf.streamDelegate.streamHasSpaceHandler = [self streamHasSpaceHandler];
+            // Start I/O event loop processing events in iAP channel
+            self.ioStreamThread = [[NSThread alloc] initWithTarget:self selector:@selector(sdl_accessoryEventLoop) object:nil];
+            [self.ioStreamThread setName:ioStreamThreadName];
+            [self.ioStreamThread start];
+        }
 
         return YES;
```

Comment:
I don't know why this function is written this way. I know it used to be this way in the first place. Can we instead do the following?

```
- (BOOL)start {
    NSString *logMessage = [NSString stringWithFormat:@"Opening EASession withAccessory:%@ forProtocol:%@", _accessory.name, _protocol];
    [SDLDebugTool logInfo:logMessage];

    self.easession = [[EASession alloc] initWithAccessory:self.accessory forProtocol:self.protocol];
    
    if (!self.easession) {
        [SDLDebugTool logInfo:@"Error: Could Not Create Session Object"];
        return NO;
    }
    
    [SDLDebugTool logInfo:@"Created Session Object"];
    
    self.streamDelegate.streamErrorHandler = [self streamErroredHandler];
    self.streamDelegate.streamOpenHandler = [self streamOpenedHandler];
    if (!self.isDataSession) {
        [self startStream:self.easession.outputStream];
        [self startStream:self.easession.inputStream];
    } else {
        self.streamDelegate.streamHasSpaceHandler = [self streamHasSpaceHandler];
        // Start I/O event loop processing events in iAP channel
        self.ioStreamThread = [[NSThread alloc] initWithTarget:self selector:@selector(sdl_accessoryEventLoop) object:nil];
        [self.ioStreamThread setName:ioStreamThreadName];
        [self.ioStreamThread start];
    }
    
    return YES;
}
```